tools/pv_tree_generator/build_tree.py

```diff
@@ -47,59 +47,90 @@ def build_tree_recursive(pos, level, pop_obs_spec, stat_vars,
     child_pos = []
     for c_pos in pop_obs_spec[level+1]:
         if (pos.pop_type == c_pos.pop_type and
-                set(pos.properties) < set(c_pos.properties)):
+                set(pos.cprops) < set(c_pos.cprops)):
             child_pos.append(c_pos)
 
-    child_prop_vals = []
-    for sv in stat_vars[pos.key]:
-        if sv.match_ui_node(prop_ui_node) and sv.pv[property_diff]:
-            if sv.pv[property_diff] not in child_prop_vals:
-                # if this is the first statsvar of this node, build the node
-                # and corresponding child nodes
-                child_prop_vals.append(sv.pv[property_diff])
-                value_ui_pv = collections.OrderedDict()
-                for prop, val in parent_pv.items():
-                    value_ui_pv[prop] = val
-                value_ui_pv[property_diff] = sv.pv[property_diff]
-                value_ui_node = util.UiNode(
-                    pos, value_ui_pv, False, property_diff)
-                value_blob = {
-                    'populationType': value_ui_node.pop_type,
-                    'sv': [sv.dcid],
-                    'l': text_format.format_title(value_ui_node.text),
-                    't': 'v',
-                    'e': value_ui_node.enum,
-                    'c': 1,
-                    'sv_set': set([sv.dcid]),
-                }
-                if sv.se:
-                    value_blob['se'] = sv.se
-                # add statistical variables as the child of current node
-                result['cd'].append(value_blob)
-
-                if level <= MAX_LEVEL:
-                    # build the branches recursively
-                    for child in child_pos:
-                        branch = build_tree_recursive(
-                            child, level + 1, pop_obs_spec, stat_vars, value_ui_node)
-                        if branch['cd']:
-                            if 'cd' not in value_blob:
-                                value_blob['cd'] = []
-                            value_blob['cd'].append(branch)
-                        value_blob['sv_set'] |= branch['sv_set']
-                        del branch['sv_set']
-                value_blob['c'] = len(value_blob['sv_set'])
-            else:
-                # if this is a duplicated statsVar
-                # append the statsVar to the right node, increase the count by 1
-                for child in result['cd']:
-                    if child['e'] == sv.pv[property_diff]:
-                        child['sv'].append(sv.dcid)
+    newLevel = (len(pos.obs_props) > 1) # create a new level if there're multiple obs_props
+    child_val_nodes = {}
+    for obs in pos.obs_props:
+        key = tuple([pos.pop_type])+ obs.key + pos.prop_all
```

Comment:
the for loop here and child_val_nodes are a bit confusing. A better way would be handle all the stats_var selection in a class.

For example, you can have brand new class called StatsVarStore, in the initialization, it will

- read all stats var
- filter duplicate stats vars
- make a private member variable indexed by the "key" here.
- expose a function like get_statsvar(pop_obs_spec, parent_pv), and return a list of stats_vars. This way can save the current line 57, 58 maybe 62, 63